Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/DownloadDiagnosisKeysTask.kt

```diff
@@ -70,18 +73,30 @@ class DownloadDiagnosisKeysTask @Inject constructor(
             checkCancel()
 
             // RETRIEVE RISK SCORE PARAMETERS
-            val exposureConfiguration = appConfigProvider.getAppConfig().exposureDetectionConfiguration
+            val exposureConfig: ExposureDetectionConfig = appConfigProvider.getAppConfig()
 
             internalProgress.send(Progress.ApiSubmissionStarted)
             internalProgress.send(Progress.KeyFilesDownloadStarted)
 
             val requestedCountries = arguments.requestedCountries
-            val availableKeyFiles = getAvailableKeyFiles(requestedCountries)
+            val keySyncResult = getAvailableKeyFiles(requestedCountries)
             checkCancel()
 
-            val totalFileSize = availableKeyFiles.fold(0L, { acc, file ->
-                file.length() + acc
-            })
+            val trackedDetections = enfClient.latestCalculations().first()
+            val now = timeStamper.nowUTC
+
+            if (wasLastDetectionPerformedRecently(now, exposureConfig, trackedDetections)) {
+                // At most one detection every 6h
+                return object : Task.Result {}
+            }
+
+            if (hasRecentDetectionAndNoNewFiles(now, keySyncResult, trackedDetections)) {
+                //  Last check was within 24h, and there are no new files.
+                return object : Task.Result {}
```

Comment:
the return statement could be wrapped in a function such as `return abortWithEmptyResult()` for readability.